maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java

```diff
@@ -190,6 +190,8 @@ public void validateFileModel( Model m, ModelBuildingRequest request, ModelProbl
             {
                 String prefix = "profiles.profile[" + profile.getId() + "].";
 
+                validateId( prefix, "id", problems, Severity.ERROR, Version.V37, profile.getId(), null, m );
```

Comment:
It has to be `Version.V40`.

Please also note that:
```
    private boolean isValidIdCharacter( char c )
    {
        return c >= 'a' && c <= 'z' || c >= 'A' && c <= 'Z' || c >= '0' && c <= '9' || c == '-' || c == '_' || c == '.';
    }
```

this needs to be tightened in the future.

Reply:
> It has to be Version.V40.
Done

> [...] needs to be tightened in the future.
I think [MNG-7052](https://issues.apache.org/jira/browse/MNG-7052) might resolve that as well.